core/payment-driver/zksync/src/zksync/wallet.rs

```diff
@@ -71,6 +71,20 @@ pub async fn init_wallet(msg: &Init) -> Result<(), GenericError> {
     Ok(())
 }
 
+pub async fn exit(address: String, msg: &Exit) -> Result<(), GenericError> {
+    let wallet = get_wallet(&address).await?;
+    let tx_handle = withdraw(wallet, msg.amount()).await?;
+    let tx_info = tx_handle
+        .wait_for_commit()
+        .await
+        .map_err(GenericError::new)?;
+    match tx_info.success {
+        Some(true) => Ok(()),
+        Some(false) => Err(GenericError::new(tx_info.fail_reason.unwrap())),
```

Comment:
Would use `unwrap_or("Unknown failure reason")`

Reply:
That's how I did it in another PR: https://github.com/golemfactory/yagna/pull/926/files#diff-932bad1dbb320024ab301b2c407b0bbddfd848c87c78f7dd2c1b44b46e7a5425